crypto/params.c

```diff
@@ -638,11 +645,14 @@ int OSSL_PARAM_set_BN(OSSL_PARAM *p, const BIGNUM *val)
 
     bytes = (size_t)BN_num_bytes(val);
     p->return_size = bytes;
+    if (sz < bytes)
```

Comment:
I think return_size should only be set on success

Reply:
It is typically set on failure too as an indication as to what space is required.

Reply:
0 is returned for multiple different types of error.
I don't think it is a good idea to assume it was a not enough space error.
If they want to figure out how much space they need - then the NULL data argument should be passed instead.

Reply:
I think that it really doesn't matter. If there's an error, the caller should act on the error, not try to second guess that maybe just maybe it was about space. There's is a mechanism for finding out the space requirement, by setting the *data* field to NULL. That doesn't change even if we do set *return_size* on error.

Granted, though, that this particular check could be positioned before the assignment of `p->return_size`

Reply:
Do we want the two `OSSL_PARAM_set_BN` calls to be inconsistent with all the rest of the set calls?

Reply:
I'll also note that by this point, the only possible error is 'not enough space'.  It seems reasonable to return the required size in this case.

All other errors set _return_size_ to zero, so they can be distinguished.

Reply:
Hmmmm, so this is an error, not so much between the OSSL_PARAM owner and the responder, but within the responder.
That's... interesting.